lib/job.js

```diff
@@ -1,106 +1,249 @@
+}
+
 /**
- * The Job class encapsulates the properties of a qless job being worked
- * on by a particular worker (our worker) and allows us to cancel the job,
- * fail the job, etc.
+ * Given a string name of a class, return the corresponding class
+ *
+ * Split the provided name by `/`, and import the longest working prefix. The
+ * remaining components will be treated as attributes off of the import.
+ *
+ * For example, consider 'some/package/file.js/property/class'. We would
+ * traverse this as:
+ *
+ *  - import 'some/package/file.js/property/class' (failed)
+ *  - import 'some/package/file.js/property', get 'class' attribute (failed)
+ *  - import 'some/package/file.js', get 'property.class' attribute (success)
  */
+Job.import = function(name) {
+  let separator = '/';
+  let segments = name.split(separator);
+  let attributes = [];
+  while (segments.length) {
+    try {
+      let joined = segments.join(separator);
+      logger.debug('Trying to import %s...', joined);
+      let result = require(joined);
+      attributes.forEach((attribute) => {
+        logger.debug('Getting attribute %s', attribute);
+        result = result[attribute];
+      })
 
-const logger = require('./util').logger('job');
-const klassFinder = require('./klass_finder');
-const errors = require('./errors');
-const _ = require('lodash');
-
-const JOB_ARGS_TO_PROPERTIES = {
-  jid: 'jid',
-  data: 'data',
-  priority: 'priority',
-  tags: 'tags',
-  state: 'state',
-  tracked: 'tracked',
-  failure: 'failure',
-  dependencies: 'dependencies',
-  dependents: 'dependents',
-  spawnedFromJid: 'spawned_from_jid',
-  expiresAt: 'expires',
-  klassName: 'klass',
-  queueName: 'queue',
-  workerName: 'worker',
-  originalRetries: 'retries',
-  retriesLeft: 'remaining',
-  rawQueueHistory: 'history',
-};
-
-
-// TODO: split up into BaseJob and RecurringJob
-class Job {
-  constructor(client, args) {
-    this.client = client;
-    this._stateChanged = false;
-
-    // Copy args
-    Object.assign(this, _.mapValues(JOB_ARGS_TO_PROPERTIES, key => args[key]));
-
-    // Some properties need extra parsing / fixing
-    this.data = JSON.parse(this.data);
-    // Side effect of Lua doing JSON parsing -- Lua empty array is {}
-    if (_.isEmpty(this.tags)) this.tags = [];
-    if (_.isEmpty(this.dependents)) this.dependents = [];
-    if (_.isEmpty(this.dependencies)) this.dependencies = [];
-  }
+      // If there's no such attribute, break out of the loop and throw
+      if (!result) {
+        break;
+      }
 
-  // If the job has been fail()ed, complete()d, etc. since object creation
-  isStateChanged() {
-    return this._stateChanged;
+      logger.debug('Found "%s"', result.name);
+      return result;
+    } catch (err) {
+      attributes.push(segments.pop());
+    }
   }
+  throw Error('Could not find ' + name);
+}
 
-  getKlass() {
-    return klassFinder.findClass(this.klassName);
-  }
+Job.prototype.setPriority = function(priority) {
+  let self = this;
```

Comment:
I don't know what I was doing wrong before, but I was able to get rid of all the `self`s. Bogons.